src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -345,6 +346,26 @@ namespace AppInstaller::Settings
             return SettingMapping<Setting::InstallLocalePreference>::Validate(value);
         }
 
+        WINGET_VALIDATE_SIGNATURE(InstallerTypePreference)
+        {
+            std::vector<Manifest::InstallerTypeEnum> installerTypes;
+            for (auto const& i : value)
+            {
+                Manifest::InstallerTypeEnum installerType = Manifest::ConvertToInstallerTypeEnum(i);
+                if (installerType == Manifest::InstallerTypeEnum::Unknown)
+                {
+                    return {};

```

Comment:
I basically followed how we validate the other settings (like Architecture preference that drop if there is an invalid value) so I left it as is because I don't think installer type should behave any differently.